nsv13/code/modules/power/stormdrive.dm

```diff
@@ -474,53 +377,280 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
+				reactor.pipe?.update_icon()
+				to_chat(usr, "<span class='notice'>Reactor outlet gate disengaged.</span>")
+				attack_hand(usr)
+				return
+			else
+				reactor.pipe?.on = TRUE
+				reactor.pipe?.target_pressure = MAX_OUTPUT_PRESSURE
+				reactor.pipe?.update_icon()
+				to_chat(usr, "<span class='notice'>Reactor outlet gate engaged.</span>")
+				attack_hand(usr)
+				return
+
+/obj/machinery/computer/ship/reactor_control_computer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state) // Remember to use the appropriate state.
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "stormdrive_console", name, 560, 600, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/ship/reactor_control_computer/ui_data(mob/user)
+	var/list/data = list()
+	data["heat"] = reactor.heat
+	data["rod_integrity"] = reactor.rod_integrity
+	data["control_rod_percent"] = reactor.control_rod_percent
+	data["pipe_open"] = reactor.pipe.on
+	data["last_power_produced"] = reactor.last_power_produced
+	data["theoretical_maximum_power"] = reactor.theoretical_maximum_power
+	data["reaction_rate"] = reactor.reaction_rate
+	if(reactor.state == REACTOR_STATE_MAINTENANCE)
+		data["reactor_maintenance"] = TRUE
+	else
+		data["reactor_maintenance"] = FALSE
+	var/moles = 0
+	if(reactor.pipe)
+		var/datum/gas_mixture/air1 = reactor.pipe.airs[1]
+		var/list/cached_gases = air1.gases
+		if(cached_gases[/datum/gas/constricted_plasma])
+			moles = cached_gases[/datum/gas/constricted_plasma][MOLES]
+			if(moles < 0)
+				moles = 0
+	data["fuel"] = moles
+	return data
+
+/obj/item/circuitboard/computer/stormdrive_reactor_control
+	name = "Stormdrive Reactor Control Console (Computer Board)"
+	build_path = /obj/machinery/computer/ship/reactor_control_computer
+
+/datum/design/board/stormdrive_reactor_control
+	name = "Computer Design (Stormdrive Reactor Control Console)"
+	desc = "Allows for the construction of circuit boards used to build a new stormdrive reactor control console."
+	id = "sd_r_c_c"
+	build_path = /obj/item/circuitboard/computer/stormdrive_reactor_control
+	category = list("Computer Boards")
+	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING
+
+/datum/techweb_node/stormdrive_reactor_control
+	id = "sd_r_c_c"
+	display_name = "Seegson RBMK RCC"
+	description = "Seegson's latest and greatest (within your budget range) reactor control design!"
+	prereq_ids = list("adv_engi", "adv_power")
+	design_ids = list("sd_r_c_c")
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+	export_price = 5000
+
+//////Reactor Manifold//////
+
+/obj/machinery/atmospherics/components/binary/pump/reactor_manifold
+	name = "Reactor Manifold"
+	desc = "The manifold connection for the stormdrive."
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
```

Comment:
I still hate myself for making this monstrosity but it'll do for the time being 🤣

Reply:
It is a time effective solution to the only issue I've seen arise from it using this method so far.
Rewriting it all is not time effective given the limited resources at our disposal.